olp-cpp-sdk-dataservice-read/tests/ParserTest.cpp

```diff
@@ -441,4 +442,108 @@ TEST(ParserTest, Index) {
+    const std::string invalid_message_json =
+        R"jsonString({"metaData":{"invalid_partition":"314010583","invalid_checksum":"ff7494d6f17da702862e550c907c0a91","invalid_compressedDataSize":152417,"invalid_dataSize":250110,"invalid_data":"iVBORw0KGgoAAAANSUhEUgAAADAAAAAwBAMAAAClLOS0AAAABGdBTUEAALGPC/xhBQAAABhQTFRFvb29AACEAP8AhIKEPb5x2m9E5413aFQirhRuvAMqCw+6kE2BVsa8miQaYSKyshxFvhqdzKx8UsPYk9gDEcY1ghZXcPbENtax8g5T+3zHYufF1Lf9HdIZBfNEiKAAAAAElFTkSuQmCC","invalid_dataHandle":"bb76b7747e7523596e74a138b15d92ec","invalid_timestamp":1517916706},"invalid_offset":{"invalid_partition":7,"invalid_offset":38562}})jsonString";
+
+    const std::string messages_json =
+        R"jsonString({"messages":[)jsonString" + valid_message_json + "," +
+        invalid_message_json + R"jsonString(]})jsonString";
+    auto messages = olp::parser::parse<olp::dataservice::read::model::Messages>(
+                        messages_json)
+                        .GetMessages();
+    ASSERT_EQ(2, messages.size());
+
+    {
+      SCOPED_TRACE("Parse valid message");
+
+      const std::string kData =
+          "iVBORw0KGgoAAAANSUhEUgAAADAAAAAwBAMAAAClLOS0AAAABGdBTUEAALGPC/"
+          "xhBQAAABhQTFRFvb29AACEAP8AhIKEPb5x2m9E5413aFQirhRuvAMqCw+"
+          "6kE2BVsa8miQaYSKyshxFvhqdzKx8UsPYk9gDEcY1ghZXcPbENtax8g5T+"
+          "3zHYufF1Lf9HdIZBfNEiKAAAAAElFTkSuQmCC";
+      const std::vector<unsigned char> kDataVec(kData.begin(), kData.end());
+      auto valid_message = messages[0];
+      // verify metadata:
+      {
+        auto metadata = valid_message.GetMetaData();
+        EXPECT_EQ(std::string("314010583"), metadata.GetPartition());
+
+        ASSERT_TRUE(metadata.GetData() != nullptr);
+        ASSERT_EQ(kDataVec, *metadata.GetData());
+
+        // Verify optional fields:
+        ASSERT_TRUE(metadata.GetChecksum());
+        ASSERT_TRUE(metadata.GetCompressedDataSize());
+        ASSERT_TRUE(metadata.GetDataSize());
+        ASSERT_TRUE(metadata.GetDataHandle());
+        ASSERT_TRUE(metadata.GetTimestamp());
+
+        ASSERT_EQ(std::string("ff7494d6f17da702862e550c907c0a91"),
+                  metadata.GetChecksum().get());
+        ASSERT_EQ(152417, metadata.GetCompressedDataSize().get());
+        ASSERT_EQ(250110, metadata.GetDataSize().get());
+        ASSERT_EQ(std::string("bb76b7747e7523596e74a138b15d92ec"),
+                  metadata.GetDataHandle().get());
+        ASSERT_EQ(1517916706, metadata.GetTimestamp().get());
+      }
+
+      // verify data:
+      ASSERT_TRUE(valid_message.GetData() != nullptr);
+      ASSERT_EQ(kDataVec, *valid_message.GetData());
```

Comment:
```suggestion
      EXPECT_EQ(kDataVec, *valid_message.GetData());
```
Consider replacing `ASSERT_*` with `EXPECT_*`